homeassistant/components/dlink/config_flow.py

```diff
@@ -19,6 +20,57 @@
 class DLinkFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for D-Link Power Plug."""
 
+    def __init__(self) -> None:
+        """Initialize a D-Link Power Plug flow."""
+        self.ip_address: str | None = None
+
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
+        """Handle dhcp discovery."""
+        await self.async_set_unique_id(discovery_info.macaddress)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: discovery_info.ip})
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if not entry.unique_id and entry.data[CONF_HOST] == discovery_info.ip:
+                # Add mac address as the unique id, can be removed with import
+                self.hass.config_entries.async_update_entry(
+                    entry, unique_id=discovery_info.macaddress
+                )
+                return self.async_abort(reason="already_configured")
+
+        self.ip_address = discovery_info.ip
+        return await self.async_step_confirm_discovery()
+
+    async def async_step_confirm_discovery(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Allow the user to confirm adding the device."""
+        errors = {}
+        if user_input is not None:
+            error = await self.hass.async_add_executor_job(
+                self._try_connect, user_input
+            )
+            if error is None:
```

Comment:
```suggestion
            if not (error := await self.hass.async_add_executor_job(
                self._try_connect, user_input
            )):
```

Not that it makes much difference but you could write it as above or `is not None:`